### ISSUE TITLE
Batch size for general dataset

### ISSUE THREAD

**Issue by ElApseR** [jsyoon0823/GAIN#6]
Hi, while reading your fabulous paper, I  just got a few question.

### Q1
For this MNIST dataset, we can use mini batch. 
But for general, matrix-shaped dataset, how did you design mini batch?
I am just guessing that the size of mini-batch might be the size of selected rows.
    i.e. X.shape=[n,p]   =>  X_mini_batch.shape=[k,p] (k<n)
Am I guessing right?

### Q2 
Is it possible that, G actually knows the full dataset?
I mean, since you are masking dataset randomly at each epoch, if we run total 10000 epoch, G might have seen total dataset without masking.
(By simple calculation, I found out that G have seen 99.99% of unmasked training dataset.)
I think for each data(X_i) in dataset(X), we should fix the mask to examine the model performance more precisely.
(And, this might be the case of the real world)
I'm bit confused.. am I wrong?
Did you also used this random-masking on your other experiments in this paper?

### Q3
I think your are using dataset matrix in vectorized form. 
For MNIST dataset, it might be okay since all variables in this data mean the same thing.
But for general dataset such as cancer... etc, each column has different meaning(also, different type)
I think simply vectorizing these datasets might cause significant loss of the information..
Was it okay for you, or did you modified G&D for those datasets?

제가 부족한 부분이 많아서 질문이 좀 많습니다, 죄송합니다..
깃헙 이슈라서 영어로 적긴 했는데, 한국말이 가능하시다면 한국말로 설명해주셔도 괜찮습니다(사실 그걸 더 선호합니다ㅎㅎ)
논문 정말 재밌게 잘 읽었습니다!

**Comment by jsyoon0823** [comment]
A1: Yes. It is right. 
- We use random permutation function to select the subset of rows with the same number of mini-batch.

A2: No we pre-determine the mask vector.
- As can be seen in line 50-51, we pre-determined the mask vector for training set and testing set. 
- Therefore, despite with multiple epochs, the generator cannot know the full dataset.

A3: Usually, the numerical data is represented to matrix formation.
- For instance, in medical data, each row represents each patient.
- Each column represents the characteristics of the patients.
- Therefore, the entire matrix has the same characteristics in a column-wise and it does not lose any information when we formalize to vectorized form.

If you have any further questions (and if you want me to answer in Korean), please email me.

Thanks.

**Comment by ElApseR** [comment]
Oh, I missed that part in your code! 
Thanks a lot for your fast reply!!
